server/internal/core/application/covenant.go

```diff
@@ -423,6 +447,11 @@ func (s *covenantService) GetInfo(ctx context.Context) (*ServiceInfo, error) {
 		return nil, err
 	}
 
+	marketNextStart := s.marketStartTime
+	if s.marketStartTime < time.Now().Unix() {
+		marketNextStart = s.marketStartTime + s.marketHourPeriod
+	}
```

Comment:
as far as i understood, `s.marketStartTime` is the very first market hour's starting time so this should be put in a loop as now it returns always the second market hour, which is not always the next ofc